pkg/iprange/parse_test.go

```diff
@@ -0,0 +1,258 @@
+package iprange
+
+import (
+	"fmt"
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseRanges(t *testing.T) {
+	tests := map[string]struct {
+		input      string
+		wantRanges []Range
+		wantErr    bool
+	}{
+		"single range": {
+			input: "192.0.2.0-192.0.2.10",
+			wantRanges: []Range{
+				prepareV4Range("192.0.2.0", "192.0.2.10"),
+			},
+		},
+		"multiple ranges": {
+			input: "2001:db8::0 192.0.2.0-192.0.2.10 2001:db8::0/126 192.0.2.0/255.255.255.0",
+			wantRanges: []Range{
+				prepareV6Range("2001:db8::0", "2001:db8::0"),
+				prepareV4Range("192.0.2.0", "192.0.2.10"),
+				prepareV6Range("2001:db8::1", "2001:db8::2"),
+				prepareV4Range("192.0.2.1", "192.0.2.254"),
+			},
+		},
+		"single invalid syntax": {
+			input:   "192.0.2.0-192.0.2.",
+			wantErr: true,
+		},
+		"multiple invalid syntax": {
+			input:   "2001:db8::0 192.0.2.0-192.0.2.10 2001:db8::0/999 192.0.2.0/255.255.255.0",
```

Comment:
please, add in the `invalid range`, a range with reverse order or change `192.0.2.0-192.0.2.10` to `192.0.2.10-192.0.2.0`.

Reply:
You also could add an invalid mask, for example, instead `192.0.2.0/255.255.255.0`, you could use `255.255.255.0/192.0.2.0`

Reply:
I saw now that the next function has the tests I am suggesting here.

Reply:
both of these cases are tested in `ParseRange`

start > end
https://github.com/ilyam8/go.d.plugin/blob/11eacd11ecb077277d33de799364d2084afd0847/pkg/iprange/parse_test.go#L79-L81

invalid/not canonical form mak

https://github.com/ilyam8/go.d.plugin/blob/11eacd11ecb077277d33de799364d2084afd0847/pkg/iprange/parse_test.go#L151-L158

---

i dont need to test these cases in the `ParseRanges`, i test only its gist - it parses single and multiple ranges (exact ranges types doesnt matter)